content/environment_variables.tex

```diff
@@ -31,13 +31,25 @@
 %%
 \EnvVarDecl{SHMEM\_SYMMETRIC\_SIZE}
     & Non-negative integer
-    & Number of bytes to allocate for symmetric heap
+    & Controls the size of the symmetric heap per \ac{PE}. The value
+      set in this environment variable is interpreted as number
+      of bytes, unless the number is followed by a single
+      character that acts as a multiplier, where:
+      \begin{itemize}
+        \item k or K multiplies by \(2^{10}\)  (kilobytes)
+        \item m or M multiplies by \(2^{20}\)  (megabytes)
+        \item g or G multiplies by \(2^{30}\)  (gigabytes)
+      \end{itemize}
+      For example, the string 20m is equivalent to the integer value 20971520,
+      or 20 megabytes. Only one multiplier is recognized, so 20kk
+      will not produce the same value as 20m, nor will invalid
+      strings such as 20MB produce the desired result.
```

Comment:
Needs an error condition. I *hate* the idea that this is implementation-defined. I want the program to error out instead of silently using the wrong symmetric heap size (or worse, the implementation picks one for you and ignores your obviously wrong size request).

"Only one multiplier is recognized, so 20kk and 20MB are not equivalent to 20m and will <generate some error>."
with e.g., <generate some error> = "terminate the program with an error report for unrecognized suffix"

Reply:
@BryantLam How about:
Any unrecognized suffix is treated as an error and results in program termination with an error report.

Reply:
fixed - recommended changes are outdated - updated with new texts - marking as resolved